app/DTO/AlbumProtectionPolicy.php

```diff
@@ -2,50 +2,71 @@
 
 namespace App\DTO;
 
-class AlbumProtectionPolicy extends DTO
-{
-	public const IS_PUBLIC_ATTRIBUTE = 'is_public';
-	public const REQUIRES_LINK_ATTRIBUTE = 'requires_link';
-	public const IS_NSFW_ATTRIBUTE = 'is_nsfw';
-	public const IS_DOWNLOADABLE_ATTRIBUTE = 'is_downloadable';
-	public const IS_SHARE_BUTTON_VISIBLE_ATTRIBUTE = 'is_share_button_visible';
-	public const GRANTS_FULL_PHOTO_ATTRIBUTE = 'grants_full_photo';
-
-	public bool $isPublic;
-	public bool $requiresLink;
-	public bool $isNSFW;
-	public bool $isDownloadable;
-	public bool $isShareButtonVisible;
-	public bool $grantsFullPhoto;
+use App\Models\BaseAlbumImpl;
+use App\Models\Extensions\BaseAlbum;
+use App\SmartAlbums\BaseSmartAlbum;
 
+/**
+ * This represents the Album Protection policy.
+ *
+ * In other words it provides information on the security attributes of an album.
+ * It MUST not be used in the front-end to determine whether an action is permitted or not (e.g. share link available).
+ *
+ * It is used in two places:
+ * 1. When sent to the front-end:
+ *   - provides an overview of the policy of the current visited album
```

Comment:
```suggestion
 *   - provides an overview of the policy of the currently visited album
```

What do you mean by "overview"? For me an overview is something like a brief summary or an abstract.  But here, the DTO sends the entire policy. What is the overview then?

Reply:
Fixed [31a5e2b](https://github.com/LycheeOrg/Lychee/pull/1539/commits/31a5e2b97c5302595e7b13326c54d6980c16498b)